_posts/en/newsletters/2019-12-04-newsletter.md

```diff
@@ -0,0 +1,155 @@
+  could improve safety and possibly eliminate the need to include
+  `OP_CODESEPARATOR` in tapscript.
+
+    Following some [IRC discussion][irc checksig pos] by several
+    participants, Anthony Towns [replied][towns checksig pos] with a
+    suggested alternative: scripts that are susceptible to this problem
+    should have their branches separated into multiple taproot leaves
+    each with just one code branch.  Tapscript signatures already commit
+    to the script being executed, so a signature that's valid for one
+    script can't be used in another script.  Towns also described why
+    committing only to position might not guarantee protection against
+    repositioned signatures and made a non-serious proposal for a method
+    that he thinks could provide superior protection, although he
+    believes it's not particularly useful compared to just keeping
+    `OP_CODESEPARATOR` in tapscript.
+
+    In a separate schnorr-related topic, ZmnSCPxj wrote a [post][zmn
+    composable musig] about the challenges of safely using the [MuSig][]
+    signature aggregation protocol with sub-groups.  For example,
+    ZmnSCPxj's [nodelets proposal][] would like to allow Alice and Bob
+    to jointly control some funds through a single LN node, using the
+    aggregation of their keys, (A, B).  Their joint node could then open
+    a channel to Charlie's node, using MuSig aggregation there too, ((A,
+    B), C).  However, ZmnSCPxj describes why this might be unsafe given
+    Wagner's algorithm as described in [last week's newsletter][news74
+    taproot updates].  Also described are several alternative schemes
+    that attempt to work around the problem.
+
+- **`OP_CHECKTEMPLATEVERIFY` (CTV):** the successor to
+  `OP_CHECKOUTPUTSHASHVERIFY` (COSHV) described in [Newsletter
+  #48][news48 coshv] and `OP_SECURETHEBAG` mentioned in [Newsletter
+  #49][news49 stb], this new opcode [proposed][ctv post] by Jeremy Rubin
+  would allow a script to ensure its funds could only be spent by a
+  specified set of child transactions.  In addition to the name change,
+  Rubin has also added additional details to the [proposed BIP][bip-ctv]
+  and he plans to hold a review workshop in the first few months of 2020
+  (fill out [this form][ctv workshop] if you're interested in
+  attending).
+
+    On-list, Russell O'Connor [restated][oconnor state variable] a
+    [previous concern][oconnor suggested amendments] of his about CTV
+    pulling data off the stack in an abnormal order for Bitcoin Script.
+    This was added by Rubin in order to prevent the creation of
+    recursive [covenants][topic covenants]---script conditions that
+    apply not just to a finite set of descendant transactions but which
+    will apply to an infinite set of descendant transactions.  For
+    example, a spender could ensure that their coins could only ever
+    again be sent to a set of three pubkeys---any payment to any other
```

Comment:
"only ever again be sent" is awkward. Maybe "only be sent to a set of three pubkeys in the future" as an alternative?